src/mod/endpoints/mod_sofia/mod_sofia.c

```diff
@@ -586,8 +586,10 @@ switch_status_t sofia_on_hangup(switch_core_session_t *session)
 
 				if (tech_pvt->respond_phrase) {
 					phrase = su_strdup(nua_handle_home(tech_pvt->nh), tech_pvt->respond_phrase);
-				} else {
+				} else if (sip_cause >= 400 && sip_cause < 700) {
```

Comment:
Why did you not include disconnect by redirection (3xx)?

Reply:
I assume that on sofia_on_hangup the possible sip range is between 4xx to 6xx. I'll correct the sip cause range.

Reply:
Fixed